vulkan/vulkan_native_surface_android.cc

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/vulkan/vulkan_native_surface_android.h"
+#include "third_party/skia/include/gpu/vk/GrVkBackendContext.h"
+
+namespace vulkan {
```

Comment:
* All classes in `lib/vulkan` are in the `vulkan` namespace.
* Classes in the `vulkan` have nothing to do with the Flutter shell and compile cleanly on any platform (It can even compile on iOS since we have our own VK headers and perform proc resolution ourselves too).
* The Flutter shell provides a generic interface (in `flutter/shell/common`) that platforms can create concrete implementations from (`flutter/shell/gpu/<client_rendering_api>`). `GPUSurfaceVulkan` is an implementation of such a concrete implementation. It uses the APIs in the `vulkan` namespace to fulfill the generic contract mandated by the Flutter Shell.